graphs/eacs/src/test/kotlin/com/anaplan/engineering/azuki/graphs/eacs/BEH6.kt

```diff
@@ -0,0 +1,100 @@
+package com.anaplan.engineering.azuki.graphs.eacs
+
+import com.anaplan.engineering.azuki.core.runner.Eac
+import com.anaplan.engineering.azuki.core.system.BEH
+import com.anaplan.engineering.azuki.graphs.adapter.api.GraphBehaviours
+import com.anaplan.engineering.azuki.graphs.adapter.api.GraphFunctions
+import com.anaplan.engineering.azuki.graphs.dsl.GraphScenario
+import com.anaplan.engineering.azuki.graphs.graphA
+
+@BEH(GraphBehaviours.HasCycles, GraphFunctions.HasCycles, """
+    Determine if a given directed graph contains one or more cycles
+""")
+class BEH6 : GraphScenario() {
+
+    @Eac("A cycle is a continuous path which starts and ends at the same vertex")
+    fun hasOneCycleDirected() {
+        given {
+            thereIsADirectedGraph(graphA) {
+                edge("a", "b")
+                edge("b", "c")
+                edge("d", "a")
+                edge("e", "d")
+                edge("c", "e")
+            }
+        }
+        then {
+            hasCycles(graphA, true)
+        }
+    }
+
+    @Eac("When there is no continuous path which starts and ends at the same vertex there are no cycles")
+    fun hasNoCyclesDirected() {
+        given {
+            thereIsADirectedGraph(graphA) {
+                edge("a", "b")
+                edge("b", "c")
+                edge("d", "a")
+                edge("e", "d")
+            }
+        }
+        then {
+            hasCycles(graphA, false)
+        }
+    }
+
+    @Eac("""
+        When the given graph is directed, any path which would be considered a cycle in an undirected graph but
+        where two edges start (or end) at the same vertex is not counted as a cycle
+        """)
+    fun hasReversedEdgeDirected() {
+        given {
+            thereIsADirectedGraph(graphA) {
+                edge("a", "b")
+                edge("b", "c")
+                edge("d", "a")
+                edge("e", "d")
+                edge("e", "c")
+            }
+        }
+        then {
+            hasCycles(graphA, false)
+        }
+    }
+
+    @Eac("When the given graph is empty, return false")
+    fun hasEmptyDirectedGraph() {
+        given {
+            thereIsADirectedGraph(graphA) {}
+        }
+        then {
+            hasCycles(graphA, false)
+        }
+    }
+
+    @Eac("When there is a continuous path that only goes to one other vertex in the given graph, a cycle exists")
```

Comment:
🚫 This is a very weird phrasing, if nothing else because it doesn't directly imply that the path you have identified is a cycle. I would suggest:

> A cycle can consist of a pair of vertices with two edges between them

Note that this is not specific to directed graphs, so thee should be no mention of edge directions here. The condition on the direction follows from the previous AC. You can include it in the notes for extra context, if you want:

```
@Eac("A cycle can consist of a pair of vertices with two edges between them",
    "If the given graph is directed, the two edges must have opposite directions.")
```